php/libraries/DataFrameworkMenu.class.inc

```diff
@@ -183,5 +183,91 @@ abstract class DataFrameworkMenu extends NDB_Menu_Filter
         return new \LORIS\Http\DataIteratorBinaryStream($rows);
 
     }
+
+    /**
+     * Format the list of user's projects to display the module filter options
+     *
+     * @param bool $numericKeys If the returned projects keys
+     *                          should be numeric (ProjectID) or project name
+     *
+     * @return array - an associative array of the form (ID|mame => name)
+     */
+    public function getProjectOptions(
+        bool  $numericKeys = true
+    ): array {
+        $projects     = \NDB_Factory::singleton()->user()->getProjects();
+        $projectNames = array_map(
+            fn($project) => $project->getName(),
+            $projects
+        );
+        if ($numericKeys) {
+            $projectIDs = array_map(
+                fn($project) => strval($project->getId()),
+                $projects
+            );
+            return array_combine($projectIDs, $projectNames);
+        } else {
+            return array_combine($projectNames, $projectNames);
+        }
+    }
+
+    /**
+     * Get the list of subprojects for which the current user has permission
+     * To populate the module subprojects filter options
+     *
+     * @param bool $numericKeys If the returned subprojects keys
+     *                          should be numeric (SubprojectID)
+     *                          or strings (subproject name)
+     *
+     * @return array - an associative array of the form (ID => name)
+     */
+    public function getSubprojectOptions(
+        bool $numericKeys = true
```

Comment:
Also don't see anywhere this is used..

Reply:
Some module filters works with index keys (half uses indexes, half names):
- modules/mri_violations/php/mri_violations.class.inc
- modules/mri_violations/php/resolved_violations.class.inc
- modules/behavioural_qc/php/behavioural.class.inc
- modules/behavioural_qc/php/conflicts.class.inc
- modules/behavioural_qc/php/incomplete.class.inc
- modules/media/php/media.class.inc
- modules/new_profile/php/new_profile.class.inc
- modules/statistics/php/stats_behavioural.class.inc
- modules/statistics/php/stats_demographic.class.inc
- modules/statistics/php/stats_mri.class.inc
- modules/user_accounts/php/edit_user.class.inc